internal/app/db/storage/postgres/order.go

```diff
@@ -0,0 +1,89 @@
+package postgres
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"log/slog"
+
+	"github.com/AsakoKabe/gophermart/internal/app/db/models"
+)
+
+type OrderStorage struct {
+	db *sql.DB
+}
+
+func NewOrderStorage(db *sql.DB) *OrderStorage {
+	return &OrderStorage{db: db}
+}
+
+const insertOrder = "insert into orders (num, user_id) values ($1, $2)"
+const selectOrderByNum = "select id, num, user_id, trim('\"' from to_json(uploaded_at)::text) from orders where num = $1"
+const selectOrdersByUserID = "select id, num, user_id, trim('\"' from to_json(uploaded_at)::text) from orders where user_id = $1 order by uploaded_at"
+
+func (s *OrderStorage) Add(ctx context.Context, order *models.Order) error {
+	_, err := s.db.ExecContext(ctx, insertOrder, order.Num, order.UserID)
+	if err != nil {
+		return fmt.Errorf("unable to insert new user: %w", err)
+	}
+
+	return nil
+}
+
+func (s *OrderStorage) GetOrderByNum(ctx context.Context, num string) (*models.Order, error) {
+	rows, err := s.db.QueryContext(
+		ctx,
+		selectOrderByNum,
+		num,
+	)
+	if err != nil {
+		slog.Error("error select order by num", slog.String("err", err.Error()))
+		return nil, err
+	}
+	defer rows.Close()
+	if !rows.Next() {
+		return nil, nil
+	}
+
+	order, err := s.parseOrder(rows)
+	if err != nil {
+		slog.Error("error to parse order", slog.String("err", err.Error()))
+		return nil, err
+	}
+
+	return order, nil
+}
+
+func (s *OrderStorage) parseOrder(rows *sql.Rows) (*models.Order, error) {
+	var order models.Order
+	if err := rows.Scan(&order.ID, &order.Num, &order.UserID, &order.UploadedAt); err != nil {
+		slog.Error("error parse order from db", slog.String("err", err.Error()))
+		return nil, err
+	}
+
+	return &order, nil
+}
+
+func (s *OrderStorage) GetOrdersByUserIDSortedByUpdatedAt(ctx context.Context, userID string) (*[]models.Order, error) {
```

Comment:
```suggestion
func (s *OrderStorage) GetOrdersByUserIDSortedByUpdatedAt(ctx context.Context, userID string) ([]*models.Order, error) {
```
Ты не первый, и я не знаю, откуда вы списываете эту странную возвращаемую конструкцию. Слайс и так является ссылочной структурой языка. А вот заполненные модели в него полезно запихивать в виде ссылок - так проще и дешевле итерировать.